application/plugins/stop_manager/controllers/Stop_manager.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * Kalkun
+ * An open source web based SMS Management
+ *
+ * @package     Kalkun
+ * @author      Kalkun Dev Team
+ * @license     http://kalkun.sourceforge.net/license.php
+ * @link        http://kalkun.sourceforge.net
+ */
+
+// ------------------------------------------------------------------------
+
+/**
+ * Stop manager Class
+ *
+ * @package     Kalkun
+ * @subpackage  Plugin
+ * @category    Controllers
+ */
+include_once(APPPATH.'plugins/Plugin_Controller.php');
+
+class Stop_manager extends Plugin_Controller {
+
+    function __construct()
+    {
+        parent::__construct();
+        $this->load->model('Kalkun_model');
+        $this->load->model('Stop_manager_model');
+    }
+
+    function index()
+    {
+        if($_POST)
+        {
+            $this->Stop_manager_model->add(
+                $this->input->post('destination_number',TRUE),
+                $this->input->post('stop_type',TRUE),
+                $this->input->post('stop_message',TRUE));
+            redirect('plugin/stop_manager');
+        }
+
+        $this->load->library('pagination');
+        $config['base_url'] = site_url('plugin/stop_manager');
+        $config['total_rows'] = $this->Stop_manager_model->get('count');
+        $config['per_page'] = $this->Kalkun_model->get_setting()->row('paging');
+        $config['cur_tag_open'] = '<span id="current">';
+        $config['cur_tag_close'] = '</span>';
+        ($this->uri->segment(3,0) == "index") ? $config['uri_segment'] = 4 : $config['uri_segment'] = 3;
+        $this->pagination->initialize($config);
+
+        $offset = ($this->uri->segment(3,0) == "index") ? $this->uri->segment(4,0) : $this->uri->segment(3,0);
```

Comment:
Is there any better way to access these parameters?

Reply:
I mostly copied what is in other parts of kalkun. I guess there isn't